### PR TITLE
Fix escape code rendering on Windows + Update typeshed main branch in doc

### PR DESCRIPTION
**Pull request by Avasam**
Demo:
image
\+ centralized the platform-specific constants under a single check.

\+ Updated the `strip_colour_code` regex:
- Standardize `\033` vs `\x1b`
- Regex matches codes more tightly

I've also updated the name of the default branch for typeshed in the doc cuz I noticed that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,12 +107,12 @@ mypy_primer -o diff
 
 See the impact of your risky change with:
 ```
-mypy_primer --repo https://github.com/hauntsaninja/mypy.git --new my_risky_change --old master
+mypy_primer --repo https://github.com/hauntsaninja/mypy.git --new my_risky_change --old main
```
Comment (hauntsaninja) [inline README.md:110]: mypy still uses master

File: mypy_primer.py [hunk 2]
```diff
@@ -1713,6 +1726,37 @@ def inner() -> int | None:
         mypy_cmd="{mypy} src",
         pip_cmd="{pip} install pydantic beartype hydra-core",
     ),
+    Project(
```
Comment (hauntsaninja) [inline mypy_primer.py:1729]: Want to add new projects in a separate PR?

Comment (Avasam) [reply mypy_primer.py:1729]: I meant to unstage that. I accidentally pushed with my second commit

**Review by hauntsaninja — APPROVED** [review]

**Comment by hauntsaninja** [comment]
Thank you!

**Pull request merged**